main.go

```diff
@@ -660,6 +744,34 @@ func handleTransmitThread(ch chan<- prometheus.Metric, threadName string, thread
 	handleIps(ch, threadName, thread)
 }
 
+func handleFlowEndCommon(ch chan<- prometheus.Metric, threadName string, flow map[string]any) {
+	// Handle flow end metrics in common between WK and FR threads.
+
+	if end, ok := flow["end"].(map[string]any); ok {
+		for _, m := range perThreadFlowEndMetrics {
+			if cm := newConstMetric(m, end, threadName); cm != nil {
+				ch <- cm
+			}
+		}
+	}
+
+	if state, ok := flow["state"].(map[string]any); ok {
```

Comment:
```suggestion
	if state, ok := end["state"].(map[string]any); ok {
```

This needs to be inside the above `if`, else `end` is not defined. Or switch it to an early return style, either is fine for me.